test_runner/fixtures/pageserver/utils.py

```diff
@@ -249,15 +249,15 @@ def assert_prefix_empty(neon_env_builder: "NeonEnvBuilder", prefix: Optional[str
             # this has been seen in the wild by tests with the below contradicting logging
             # https://neon-github-public-dev.s3.amazonaws.com/reports/pr-5322/6207777020/index.html#suites/3556ed71f2d69272a7014df6dcb02317/53b5c368b5a68865
             # this seems like a mock_s3 issue
-            log.warn(
+            log.warning(
                 f"contrading ListObjectsV2 response with KeyCount={keys} and Contents={objects}, CommonPrefixes={common_prefixes}, assuming this means KeyCount=0"
             )
             keys = 0
         elif keys != 0 and len(objects) == 0:
             # this has been seen in one case with mock_s3:
             # https://neon-github-public-dev.s3.amazonaws.com/reports/pr-4938/6000769714/index.html#suites/3556ed71f2d69272a7014df6dcb02317/ca01e4f4d8d9a11f
             # looking at moto impl, it might be there's a race with common prefix (sub directory) not going away with deletes
-            log.warn(
+            log.warning(
```

Comment:
I've yet to see any such local warnings. Do you have an example?

Reply:
They come at the end of a test run, like this:
```
[gw3] [ 40%] PASSED test_runner/regress/test_pageserver_secondary.py::test_live_migration[release-pg15]
[gw1] [ 60%] PASSED test_runner/regress/test_pageserver_secondary.py::test_location_conf_churn[release-pg15-2]
[gw0] [ 80%] PASSED test_runner/regress/test_pageserver_secondary.py::test_location_conf_churn[release-pg15-1]
[gw2] [100%] PASSED test_runner/regress/test_pageserver_secondary.py::test_location_conf_churn[release-pg15-3]

=============================== warnings summary ===============================
test_runner/regress/test_pageserver_secondary.py::test_secondary_downloads[release-pg15]
  /home/neon/neon/test_runner/fixtures/pageserver/utils.py:252: DeprecationWarning: The 'warn' method is deprecated, use 'warning' instead
    log.warn(

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=================== 5 passed, 1 warning in 101.48s (0:01:41) ===================
```

Reply:
Ok, interesting, I don't see them with python3.9. Makes sense to switch. Sadly you did not see the actual warning, which has been visible in ci only one or two times.